### PR TITLE
Hierarchically Deterministic Node IDs

### PR DESCRIPTION
**Pull request by braydonf**
Reference Implementation: 
- https://github.com/Storj/core/pull/527
- https://github.com/Storj/complex/pull/13
- https://github.com/Storj/bridge/pull/260

### CODE REVIEW

File: sip-hdnodeid.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+```
+SIP: hdnodeid
```
Comment (161chihuahuas) [inline sip-hdnodeid.md]: Is it normal to use a short name like this or the numerical id?

**Review by 161chihuahuas — COMMENTED** [review]
Be sure to add a link to the README in the little index table too

File: sip-hdnodeid.md (hunk 1 above)
Comment (braydonf) [reply sip-hdnodeid.md]: The numerical id hasn't been assigned yet, so it's a placeholder. Need an sip number, and I'll update, and add to the readme.

Comment (super3) [reply sip-hdnodeid.md]: @bookchin Per SIP 0001 you are responsible for assigning SIP numbers. In the numbering you should leave some space after this SIP as we may want to update it with another SIP in the future.

**Comment by super3** [comment]
@braydonf Note we typically we pre-pend working pull requests with [WIP].

**Comment by super3** [comment]
@braydonf I think the key tree diagram we had on the board would be a super helpful for those reading this SIP, and should be included. Can help you with that on Monday if you would like.

File: sip-hdnodeid.md (hunk 1 above)
Comment (161chihuahuas) [reply sip-hdnodeid.md]: @braydonf okay, well so far we've got `0001`, `0002`, `0003`, so obviously this one gets to be `0032`. :stuck_out_tongue_closed_eyes:

Comment (161chihuahuas) [reply sip-hdnodeid.md]: But seriously.

**Pull request merged**